### PR TITLE
[flutter_tools] measure driver success and failure

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Currently we only have a test that measures if flutter drive can fail. Unfortunately it has been failing for multiple years based on the fact that the error is due to a missing Gradle config.

Add a smoke success test case, and update both tests to use the test API and the tester-device.

### CODE REVIEW

File: packages/flutter_driver/test_driver/success_test.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import '../test/common.dart';
+
+void main() {
+  // Tooling requires a single trivial test
+  test('it runs a test', () {
+    expect(true, true);
```
Comment (yjbanov) [inline packages/flutter_driver/test_driver/success_test.dart:12]: do we need this line?

**Review by yjbanov — APPROVED** [review]
lgtm

File: packages/flutter_driver/test_driver/success_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_driver/test_driver/success_test.dart:12]: I want to actually run a test to make sure the `pub run test` logic works

Comment (jonahwilliams) [reply packages/flutter_driver/test_driver/success_test.dart:12]: oh, you mean the expect? I guess not ... but a totally empty test seemed weird

**Pull request merged**